src/components/TabSelector/TabSelector.js

```diff
@@ -62,32 +62,43 @@ const getOpacity = (position, routesLength, tabIndex, active) => {
 
         return position.interpolate({
             inputRange,
-            outputRange: _.map(inputRange, (i) => (i === tabIndex ? activeValue : inactiveValue)),
+            outputRange: _.map(inputRange, (i) => (affectedTabs.includes(tabIndex) && i === tabIndex ? activeValue : inactiveValue)),
         });
     }
     return activeValue;
 };
 
-const getBackgroundColor = (position, routesLength, tabIndex) => {
+const getBackgroundColor = (position, routesLength, tabIndex, affectedTabs) => {
     if (routesLength > 1) {
         const inputRange = Array.from({length: routesLength}, (v, i) => i);
 
         return position.interpolate({
             inputRange,
-            outputRange: _.map(inputRange, (i) => (i === tabIndex ? themeColors.border : themeColors.appBG)),
+            outputRange: _.map(inputRange, (i) => (affectedTabs.includes(tabIndex) && i === tabIndex ? themeColors.border : themeColors.appBG)),
         });
     }
     return themeColors.border;
 };
 
 function TabSelector({state, navigation, onTabPress, position}) {
     const {translate} = useLocalize();
+
+    const defaultAffectedAnimatedTabs = Array.from({length: state.routes.length}, (v, i) => i);
+    const [affectedAnimatedTabs, setAffectedAnimatedTabs] = useState(defaultAffectedAnimatedTabs);
+
+    React.useEffect(() => {
+        setTimeout(() => {
```

Comment:
We are usually against using setTimeout, why do we need it here?

Reply:
We need here to reset `affectedAnimatedTabs` after transition end, and I think this is the available way to wait transition end.

Removing `setTimeout` will back the issue to happened again because while moving from tab1 to tabs3 the transition is not end yet and at this time calling reset `affectedAnimatedTabs` to default will reset it to [tab1, tab2, tab3] while we need it to keep it [tab1, tab3] until transition end

Reply:
Got it! Can you leave a comment then explaining why?

Reply:
@pecanoro updated.